receiver/elasticapmreceiver/receiver.go

```
@@ -236,18 +246,59 @@ func (r *elasticAPMReceiver) processBatch(ctx context.Context, batch *modelpb.Ba
 			ss := rs.ScopeSpans().AppendEmpty()
 			s := ss.Spans().AppendEmpty()
 
+			traceId, err := traceIDFromHex(event.Trace.Id)
```

Is there a good reason not to log an error here (and below)? An issue in the format would be silently ignored, wrong values propagate into production and are possibly hard to recognize or debug.

---

This is a good point, thanks. We should log this indeed.

I added error level logs and also added one more test case to cover when parsing fails.

Other (in my opinion less good) options I see other than the error logs are:
- just log as warning: I think missing trace id, span id, and parent id is more than just a warning. At least error is warranted because such traces won't be very useful - e.g. correlation won't work and similar
- return with an error and don't map the event: I think this would be too brutal and it'll result by completely missing events. There is nothing after this point in the pipeline that'd prevent the empty ids, so we'd still have all the other information. I feel keeping what's possible is better for users.

So from that reason, I just added an error log as suggested and let this thing happen.